3scaleAdapter/pkg/threescale/threescale.go

```diff
@@ -64,73 +64,74 @@ var _ logentry.HandleLogEntryServiceServer = &Threescale{}
 
 func (s *Threescale) HandleLogEntry(ctx context.Context, l *logentry.HandleLogEntryRequest) (*v1beta1.ReportResult, error) {
 	var report v1beta1.ReportResult
-	var err error
 
-	if l.AdapterConfig == nil {
-		return &report, fmt.Errorf("adapter config not available")
-	}
+	go func() {
```

Comment:
So this is going to spawn new goroutine for every call?

Is it ok with 3000 rps? Is it ok when backend goes down? It is ok when the network slows down?

IMO the only correct way to do this is an in-memory queue and N workers that are taking that queue and reporting it.
Otherwise, you are going to spawn an unbounded number of goroutines and that can't be healthy.

Reply:
isn't the Go runtime meant to be able to spawn a bazillion goroutines?

Reply:
As any unbounded growth, it will affect scheduling, memory, etc. right?

I'm sure you can have thousands of them. Can you have tens of thousands, hundreds, millions?

Reply:
sure, it might however already implement some sort of throttling - what happens when you reach the max number of goroutines?

Reply:
Yes, I wanted to raise that point. it's unbounded... so it's dangerous. We can implement this using a limited number of workers if we hit some problem.

Reply:
> Future self. sorry.